lesson15/main.js

```diff
@@ -0,0 +1,98 @@
+"use strict";
+const wrapper = document.getElementById("js-wrapper");
+const ul = document.getElementById("js-ul");
+const submitBtn = document.getElementById("js-submitBtn");
+const modal = document.getElementById("js-modal");
+const openModalBtn = document.getElementById("js-openModalBtn");
+const jsonURL = "https://myjson.dit.upm.es/api/bins/j16z";
+const removeModal = () => {
+   modal.remove();
+   wrapper.classList.remove("bg_gray");
+};
+
+const startLoading = () => {
+   const loadingIcon = document.createElement("img");
+   loadingIcon.id = "loadingIcon";
+   loadingIcon.src = "img/loading-circle.gif";
+   wrapper.appendChild(loadingIcon);
+};
+
+const endLoading = () => {
+   const loadingIcon = document.getElementById("loadingIcon");
+   loadingIcon.remove();
+}
+
+openModalBtn.addEventListener("click", function () {
+   modal.style.display = "block";
+   wrapper.classList.add("bg_gray");
+   openModalBtn.remove();
+});
+
+const isEmptyString = (value) => value === "";
+
+submitBtn.addEventListener("click", () => {
+   const inputNumber = document.getElementById("js-number");
+   const inputText = document.getElementById("js-text");
+   if (isEmptyString(inputNumber.value) || isEmptyString(inputText.value)) { //inputNumber.valueとinputText.valueのデータ型がStringのため
+      alert("未入力です。");
+      return;
+   }
+   console.log(`ID:${inputNumber.value}`);
+   console.log(`ユーザー名:${inputText.value}`);
+   tryCreateElement();
+});
+
+async function init() {
+   removeModal();
+   startLoading();
+   try {
+      return await fetchData();
+   } catch (e) {
+      throw new Error(error);
+   } finally {
+      endLoading();
+   }
+}
+
+async function fetchData() {
+   try {
+      const response = await fetch(jsonURL);
+      const json = await response.json();
+      return json.data;
+   } catch (error) {
+      throw new Error(error);
+   }
+}
```

Comment:
I don't think we need try...catch here. The reason is that we do the same thing in init.

Reply:
Thank you for your comment.
I hadn't noticed that.
I removed that try...catch clause.
e73ccb4